### PR TITLE
Update check_chromium_src script

### PR DESCRIPTION
**Pull request by mkarolin**
Resolves https://github.com/brave/brave-browser/issues/33338

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by emerick — APPROVED** [review]
Amazing cleanup, Max!

**Review by AlexeyBarabash — APPROVED** [review]
sync ++

**Comment by fmarier** [comment]
@mkarolin I marked both warnings as safe. Should be good to go.

File: tools/chromium_src/check_chromium_src.py [hunk 1]
```diff
@@ -0,0 +1,543 @@
...
+                        help='Build type (i.e. Component|Static|Debug|Release)',
+                        nargs='?',
+                        default='Component')
+    parser.add_argument('--os',
+                        required=False,
+                        help='Target OS (e.g. android|win|linux|mac|ios)')
+    parser.add_argument('--arch',
+                        required=False,
+                        help='Target architecture (e.g. x86|x64|arm|arm64)')
+    options = parser.parse_args(args)
```
Comment (goodov) [inline tools/chromium_src/check_chromium_src.py:516]: do we really need to pass `args` here? it should work without this.
https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.parse_args

File: check_chromium_src_config.json5 [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            "BRAVE_CHROMIUM_SRC_COMPONENTS_PERMISSIONS_PERMISSION_REQUEST_DATA_PERMISSION_CONTEXT_BASE"
+        ],
+        "sandbox/win/src/interceptors_64.h": [
+            "TargetGetModuleFileNameExA64",
+            "TargetGetModuleFileNameExW64"
+        ],
+        "v8/src/builtins/builtins-utils.h": [
+            "BUILTIN"
+        ]
+    }
```
Comment (goodov) [inline check_chromium_src_config.json5:129]: everything here should end with a trailing comma for a cleaner git diff in future

File: check_chromium_src_config.json5 [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+}
```
Comment (goodov) [inline check_chromium_src_config.json5:130]: new line at end

File: chromium_src/PRESUBMIT.py [hunk 4]
```diff
@@ -44,3 +45,27 @@ def CheckOverriddenHeadersDeclareIWYUExport(input_api, output_api):
...
+def CheckOverrides(input_api, output_api):
+    items = []
+    with import_inline.sys_path(
+            input_api.os_path.join(input_api.PresubmitLocalPath(), '..',
+                                   'tools')):
+        # pylint: disable=import-error,import-outside-toplevel
+        from chromium_src import check_chromium_src
+        overrides = [
+            f.AbsoluteLocalPath() for f in input_api.AffectedSourceFiles(None)
+            if f.IsTestableFile()
```
Comment (goodov) [inline chromium_src/PRESUBMIT.py:59]: I think `AffectedSourceFiles` already includes this check.

**Review by goodov — APPROVED** [review]

**Review by cdesouza-chromium — APPROVED** [review]
Thanks for taking time to look into this.

**Comment by goodov** [comment]
you can also run `npm run presubmit -- --files=chromium_src*` and fix other issues left there :)

File: check_chromium_src_config.json5 [hunk 5]
```diff
@@ -0,0 +1,130 @@
+{
```
[the remark this answers is not in the document]
Comment (mkarolin) [reply check_chromium_src_config.json5]: Done in 9633e750d130748b0459e6d4b1fe707f5597db73

File: tools/chromium_src/check_chromium_src.py (hunk 1 above)
Comment (mkarolin) [reply tools/chromium_src/check_chromium_src.py]: Yep, not needed. Fixed in dde637633ee8265bb9dbc7a7b39cc41c4c46d9a1

File: check_chromium_src_config.json5 (hunk 2 above)
Comment (mkarolin) [reply check_chromium_src_config.json5]: Fixed in dde637633ee8265bb9dbc7a7b39cc41c4c46d9a1

File: check_chromium_src_config.json5 (hunk 3 above)
Comment (mkarolin) [reply check_chromium_src_config.json5]: Fixed in dde637633ee8265bb9dbc7a7b39cc41c4c46d9a1

File: chromium_src/PRESUBMIT.py (hunk 4 above)
Comment (mkarolin) [reply chromium_src/PRESUBMIT.py:59]: It looks like if you pass a filter into `AffectedSourceFiles` then it does return only testable files, but with `None` (falling back onto the default FilterSourceFile) it doesn't seem to apply testable check.

File: chromium_src/PRESUBMIT.py [hunk 6]
```diff
@@ -44,3 +45,27 @@ def CheckOverriddenHeadersDeclareIWYUExport(input_api, output_api):
...
+
+
+def CheckOverrides(input_api, output_api):
+    items = []
+    with import_inline.sys_path(
+            input_api.os_path.join(input_api.PresubmitLocalPath(), '..',
+                                   'tools')):
+        # pylint: disable=import-error,import-outside-toplevel
+        from chromium_src import check_chromium_src
+        overrides = [
```
[the remark this answers is not in the document]
Comment (mkarolin) [reply chromium_src/PRESUBMIT.py:57]: Fixed in dde637633ee8265bb9dbc7a7b39cc41c4c46d9a1

**Comment by mkarolin** [comment]
@goodov, please take a look at the additional fixes for issues discovered in `npm run presubmit -- --files=chromium_src*`.
I didn't fix all issues, though.

File: chromium_src/PRESUBMIT.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (goodov) [reply chromium_src/PRESUBMIT.py:57]: actually `overrides = [ ... ]` should also work without `with import_inline.sys_path(`

File: chromium_src/PRESUBMIT.py (hunk 4 above)
Comment (goodov) [reply chromium_src/PRESUBMIT.py:59]: > It looks like if you pass a filter into `AffectedSourceFiles` then it does return only testable files, but with `None` (falling back onto the default FilterSourceFile) it doesn't seem to apply testable check.

hm. what I see:
[`input_api.AffectedSourceFiles`](https://chromium.googlesource.com/chromium/tools/depot_tools.git/+/refs/heads/main/presubmit_support.py#809) calls [`input_api.AffectedTestableFiles`](https://chromium.googlesource.com/chromium/tools/depot_tools.git/+/refs/heads/main/presubmit_support.py#751) (no matter if `FilterSourceFile` is applied or not). `input_api.AffectedTestableFiles` always includes `x.IsTestableFile` check.

maybe I'm missing something?

**Review by goodov — APPROVED** [review]
lgtm! (with few comments)

**Comment by mkarolin** [comment]
Fixed the latest comments, rebased on master and squashed some commits.

**Comment by mkarolin** [comment]
`EphemeralStorageForgetByDefaultBrowserTest.NavigationCookiesAreCleared` is failing on Linux. We already have this test disabled on MacOS (https://github.com/brave/brave-browser/issues/31806), the failure on Linux looks the same. This failure is unrelated to the PR.

**Pull request merged**